src/strategy.rs

```diff
@@ -261,15 +271,400 @@ pub fn hostname<S: BuildHasher>(parameters: Option<HashMap<String, String, S>>)
     Box::new(move |_: &Context| -> bool { result })
 }
 
+/// returns true if the strategy should be delegated to, false to disable
+fn _compile_constraint_string<F>(expression: ConstraintExpression, getter: F) -> Evaluate
+where
+    F: Fn(&Context) -> Option<&String> + Clone + Sync + Send + 'static,
+{
+    match &expression {
+        ConstraintExpression::In(values) => {
+            let as_set: HashSet<String> = values.iter().cloned().collect();
+            Box::new(move |context: &Context| {
+                getter(context).map(|v| as_set.contains(v)).unwrap_or(false)
+            })
+        }
+        ConstraintExpression::NotIn(values) => {
+            if values.is_empty() {
+                Box::new(|_| true)
+            } else {
+                let as_set: HashSet<String> = values.iter().cloned().collect();
+                Box::new(move |context: &Context| {
+                    getter(context)
+                        .map(|v| !as_set.contains(v))
+                        .unwrap_or(false)
+                })
+            }
+        }
+    }
+}
+
+fn _ip_to_vec(ips: &[String]) -> Vec<ipaddress::IPAddress> {
+    let mut result = Vec::new();
+    for ip_str in ips {
+        let ip_parsed = ipaddress::IPAddress::parse(ip_str.trim());
+        if let Ok(ip) = ip_parsed {
+            result.push(ip);
+        }
```

Comment:
Should invalid IP addresses just be discarded? At the very least maybe log them with `log::debug!`?

Reply:
yah, we can; got a bit sloppy there :P. We can't error sensibly because the error would be thrown to the API server polling function, and someone updating unleash could just break every strategy across the board; a bad IP is effectively a guaranteed mismatch, thus dropping it is equivalent for the IN case